### PR TITLE
Minor follow up fixes for python registration

### PR DESCRIPTION
**Pull request by anjali411**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #76892

### CODE REVIEW

File: torch/csrc/utils/python_dispatch.cpp [hunk 1]
```diff
@@ -55,7 +55,7 @@ inline torch::CppFunction dispatch_str(const char* key, Func&& raw_f) {
 }
 
 class PythonKernelHolder : public c10::OperatorKernel {
-  SafePyObject func_;
+  c10::SafePyObject func_;
```
Comment (albanD) [inline torch/csrc/utils/python_dispatch.cpp:58]: WHy is that needed now?

File: torch/library.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ class Library:
 
     Args:
         ns: library name
-        kind: "IMPL" by default
+        kind: "DEF", "IMPL" (default: "IMPL")
```
Comment (albanD) [inline torch/library.py:23]: Note that my comment from the original PR still stands: we create the C++ object beofre checking that kind is valid!

Comment (anjali411) [reply torch/library.py:23]: yes updated!

**Review by albanD — APPROVED** [review]
I think the doc could use some cleanup but accepting to unblock.
